h5py/_hl/files.py

```diff
@@ -288,19 +294,21 @@ def atomic(self, value):
 
     if swmr_support:
         @property
+        @with_phil
         def swmr_mode(self):
             """ Controls single-writer multiple-reader mode """
-            return self._swmr_mode
+            return 0 < (self.id.get_intent() & (h5f.ACC_SWMR_READ | h5f.ACC_SWMR_WRITE))
```

Comment:
```suggestion
            return bool(self.id.get_intent() & (h5f.ACC_SWMR_READ | h5f.ACC_SWMR_WRITE))
```

Seems a bit clearer. The comparison is backwards to my way of thinking. ;-)